File: backend/api/src/PetCare.Domain/Entities/AppointmentSlot.cs
using PetCare.Domain.Common;
using PetCare.Domain.Enums;

namespace PetCare.Domain.Entities;

/// <summary>
/// A bookable time slot owned by a veterinarian.
/// See docs/database/scheduling-billing-approval-domain-model.md#appointmentslot.
/// UNIQUE (VeterinarianId, Date, StartTime) is enforced via EF Core configuration.
/// </summary>
public class AppointmentSlot : AuditableEntity
{
    public Guid VeterinarianId { get; set; }

    public Veterinarian Veterinarian { get; set; } = null!;

    public DateOnly Date { get; set; }

    public TimeOnly StartTime { get; set; }

    public TimeOnly EndTime { get; set; }

    public string Branch { get; set; } = string.Empty;

    public AppointmentSlotStatus Status { get; set; } = AppointmentSlotStatus.Available;

    /// <summary>
    /// Set when this slot is consumed by an appointment (1:1).
    /// </summary>
    public Appointment? Appointment { get; set; }
}

File: frontend/mobile/lib/features/scheduling/models/appointment_slot.dart
class AppointmentSlot {
  final String id;
  final String veterinarianId;
  final String date;
  final String startTime;
  final String endTime;
  final String branch;
  final String status;

  AppointmentSlot({
    required this.id,
    required this.veterinarianId,
    required this.date,
    required this.startTime,
    required this.endTime,
    required this.branch,
    required this.status,
  });

  factory AppointmentSlot.fromJson(Map<String, dynamic> json) {
    return AppointmentSlot(
      id: json['id'] as String,
      veterinarianId: json['veterinarianId'] as String,
      date: json['date'] as String,
      startTime: json['startTime'] as String,
      endTime: json['endTime'] as String,
      branch: json['branch'] as String,
      status: json['status'] as String,
    );
  }
}

class Appointment {
  final String id;
  final String petId;
  final String veterinarianId;
  final String appointmentSlotId;
  final String scheduledStart;
  final String scheduledEnd;
  final String status;
  final String? notes;
  final String createdAt;
  final String updatedAt;

  Appointment({
    required this.id,
    required this.petId,
    required this.veterinarianId,
    required this.appointmentSlotId,
    required this.scheduledStart,
    required this.scheduledEnd,
    required this.status,
    this.notes,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['id'] as String,
      petId: json['petId'] as String,
      veterinarianId: json['veterinarianId'] as String,
      appointmentSlotId: json['appointmentSlotId'] as String,
      scheduledStart: json['scheduledStart'] as String,
      scheduledEnd: json['scheduledEnd'] as String,
      status: json['status'] as String,
      notes: json['notes'] as String?,
      createdAt: json['createdAt'] as String,
      updatedAt: json['updatedAt'] as String,
    );
  }
}
